### PR TITLE
add support for VS Code tasks to do build and upload 

### PR DESCRIPTION
**Pull request by bugobliterator**
First step in getting betting integration of ardupilot with VSCode. The extension being developed for this:
https://github.com/ardupilot/ardupilot_vscode_devenv

Few Screenshots:

Screenshot 2021-01-06 at 4 21 30 AM
Screenshot 2021-01-06 at 4 21 50 AM
Screenshot 2021-01-06 at 4 22 11 AM
Screenshot 2021-01-06 at 4 22 31 AM
Screenshot 2021-01-06 at 4 22 36 AM

### CODE REVIEW

File: Tools/AP_Periph/wscript [hunk 1]
```diff
@@ -2,9 +2,12 @@
 # encoding: utf-8
 
 import fnmatch
+import os
+import sys
 
 def build(bld):
-    targets = ['f103-*', 'f303-*', 'CUAV_GPS', 'ZubaxGNSS*', '*-periph', 'sitl_*', 'HitecMosaic*']
+    with open(os.path.join(sys.path[0], "../AP_Periph/targets.txt"), "r") as f:
```
Comment (tridge) [inline Tools/AP_Periph/wscript:9]: there is bld.srcpath or a __FILE__ equivalent

File: Tools/AP_Periph/wscript [hunk 2]
```diff
@@ -2,9 +2,12 @@
...
+        targets = f.read().split('\n')
```
Comment (tridge) [inline Tools/AP_Periph/wscript:10]: readlines ?

File: wscript [hunk 3]
```diff
@@ -399,6 +404,40 @@ def collect_dirs_to_recurse(bld, globs, **kw):
 def list_boards(ctx):
     print(*boards.get_boards_names())
 
+def generate_tasklist(ctx, do_print=True):
+    boardlist = boards.get_boards_names()
+    tasks = []
+    with open(os.path.join(sys.path[0], "../../tasklist.json"), "w") as tlist:
+        for board in boardlist:
+            task = {}
+            with open(os.path.join(sys.path[0], "../AP_Periph/targets.txt"), "r") as periphtargets:
```
Comment (tridge) [inline wscript:413]: same issues

**Review by tridge — CHANGES_REQUESTED** [review]

File: Tools/AP_Periph/wscript (hunk 1 above)
Comment (khancyr) [inline Tools/AP_Periph/wscript:9]: isn't just make a get_targets() function simpler ?

File: wscript [hunk 4]
```diff
@@ -399,6 +405,41 @@ def collect_dirs_to_recurse(bld, globs, **kw):
...
+                    # we only support AP_Periph and bootloader builds
+                    task['targets'] = ['AP_Periph', 'bootloader']
+                else:
+                    task['targets'] = ['AP_Periph']
+            elif 'iofirmware' in board:
+                task['targets'] = ['iofirmware', 'bootloader']
+            else:
+                if 'linux' in board:
+                    # we only support AP_Periph and bootloader builds
+                    task['targets'] = ['antennatracker', 'copter', 'heli', 'plane', 'rover', 'sub', 'bootloader', 'replay']
```
Comment (tridge) [inline wscript:434]: replay is not linux any more, it's normally sitl but can do any board. Maybe list as only sitl

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**